R/learner_torch_methods.R

```diff
@@ -286,6 +286,13 @@ encode_prediction_default = function(predict_tensor, predict_type, task) {
   response = prob = NULL
   if (task$task_type == "classif") {
     if (predict_type == "prob") {
+      if (dim(predict_tensor)[2] == 1) {
+        N_data <- length(predict_tensor)
+        vec_dim <- c(N_data, 1)
+        pos_scores <- predict_tensor$reshape(vec_dim)
+        neg_scores <- torch::torch_zeros(N_data)$reshape(vec_dim)
```

Comment:
for justification of using zero for negative scores before putting them through softmax, see my slides, section "Multi-class classification" https://github.com/tdhock/2023-08-deep-learning/blob/main/slides/torch-part1/06-backprop.pdf
![image](https://github.com/user-attachments/assets/ec75e948-6188-4033-a026-324d46873096)